src/main/java/org/opensearch/flowframework/model/WorkflowState.java

```diff
@@ -69,7 +78,7 @@ public WorkflowState(
         Instant provisionEndTime,
         User user,
         Map<String, Object> userOutputs,
-        Map<String, Object> resourcesCreated
+        List<ResourcesCreated> resourcesCreated
```

Comment:
So I think I answered my previous question on more than one resource; it's handled here.  But in that case:
 - we should probably name the other class `ResourceCreated`
 - we still should have some sort of type identifier on what sort of resource it is (index, connector, etc.).  An enum is probably fine for that.

Reply:
yes will change to `ResourceCreated`. Regarding adding some variable for each step, Tyler also mentioned that with additional section for associated resource (resource we modify but don't create (knn index can be a common one with adding a pipeline to it), so users can keep track of that). 

I think it would make sense to add that in additional PR, I mentioned it in the description with the issue I have for it. I can add a TODO as well